helm/k8s-deployment-scaler/templates/deployment.yaml

```diff
@@ -0,0 +1,37 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "k8s-deployment-scaler.fullname" . }}
+  labels:
+    {{- include "k8s-deployment-scaler.labels" . | nindent 4 }}
+spec:
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 25%
+      maxSurge: 25%
+  {{- if .Values.replicaCount }}
+  replicas: {{ .Values.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "k8s-deployment-scaler.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "k8s-deployment-scaler.selectorLabels" . | nindent 8 }}
+    spec:
+      serviceAccountName: {{ include "k8s-deployment-scaler.fullname" . }}
+      containers:
+      - name: {{ .Chart.Name }}
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: IfNotPresent
+        ports:
+        - containerPort: 8443
+        readinessProbe:
+          tcpSocket:
+            port: 8443
```

Comment:
Readiness probe is TCP based on HTTPS port, as results logs are full of:

```
2024/06/27 14:11:48 http: TLS handshake error from 10.244.0.1:36052: EOF
2024/06/27 14:11:53 http: TLS handshake error from 10.244.0.1:36054: EOF
2024/06/27 14:11:58 http: TLS handshake error from 10.244.0.1:60312: EOF
2024/06/27 14:12:03 http: TLS handshake error from 10.244.0.1:60326: EOF
2024/06/27 14:12:08 http: TLS handshake error from 10.244.0.1:51634: EOF
```

You should either handle EOF from health-checker, or come up with a different way to check health.

Reply:
I will suppress the EOF error to avoid adding additional complexity or adding any non-mTLS endpoints.